ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activityEmailInput.component.ts

```diff
@@ -66,63 +73,82 @@ export class ActivityEmailInput implements OnChanges {
-        } else if (allFieldsTouched) {
-            this.saveAnswer(null);
-        }
-    }
-
     public fieldHasError(field: string, error: string): boolean {
-        return this.emailForm.touched && this.emailForm.controls[field].errors && this.emailForm.controls[field].errors[error];
+        return this.emailForm && this.emailForm.touched && this.emailForm.controls[field].errors
+            && this.emailForm.controls[field].errors[error];
     }
 
     private initEmailForm(): void {
+        interface EmailForm {
+            email: string;
+            confirmEmail?: string;
+        }
+        // can't use the Angular email validator as does not match Pepper back-end email validator
+        const EmailPatternValidator = Validators.pattern(EMAIL_REGEXP);
+
         if (this.block.confirmEntry) {
+
             this.emailForm = this.formBuilder.group({
                 email: new FormControl({
                     value: this.block.answer,
                     disabled: this.readonly
-                }, Validators.email),
+                }, EmailPatternValidator),
                 confirmEmail: new FormControl({
-                    value: this.block.answer,
+                    value: this.block.confirmationValue,
                     disabled: this.readonly
-                }, Validators.email)
+                }, EmailPatternValidator)
             }, {
-                validator: fieldsMatcher('email', 'confirmEmail')
+                validator: this.fieldsMatcher('email', 'confirmEmail')
             });
         } else {
             this.emailForm = new FormGroup({
                 email: new FormControl({
                     value: this.block.answer,
                     disabled: this.readonly
-                }, Validators.email)
+                }, EmailPatternValidator)
             });
         }
-    }
 
-    private saveAnswer(answer: string | null): void {
-        this.block.answer = answer;
-        this.valueChanged.emit(answer);
+        this.subscription = this.emailForm.valueChanges.pipe(
+            map((formData) => {
+                const cleanedFormData: EmailForm = { email: '' };
+                ['email', 'confirmEmail'].forEach(propName =>
+                    cleanedFormData[propName] = formData[propName] ? formData[propName].trim() : null
+                );
+                return cleanedFormData;
+            }),
+            tap((cleanedFormData) => {
+                this.block.setAnswer(cleanedFormData.email, false);
+                this.block.confirmEntry && (this.block.confirmationValue = cleanedFormData.confirmEmail);
+            }),
+            // we only emit data we know server will accept
+            filter((_) => this.block.canPatch()),
+            map((cleanedFormData) => !!cleanedFormData.email ? cleanedFormData.email : ''),
+            distinctUntilChanged(),
+            tap((val) => this.valueChanged.emit(val))
+        ).subscribe();
     }
```

Comment:
Big change here it to handle the value changes from reactive pipeline rather than from an onChange handler.
We get the raw changes, clean them up, store them in the block, call the block validation and emit if data looks valid.
Note that we send valid emails or empty strings.
Also noteworthy is that we store the confirmationValue in the block. We do this because our pre-existing validation scheme has validators working off data values in block, so that is why we extended the text block and set the confirmationValue; to allow the EmailValidatorRule to figure out if our email is valid or not.